frontend/src/pages/Login.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react'
+import { batch, useDispatch, useSelector } from 'react-redux' 
+import { useHistory } from 'react-router-dom'
+
+import user from '../reducers/user'
+import { API_URL } from '../reusable/urls'
+
+const Login = () => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [mode, setMode] = useState(null)
+  
+  const errors = useSelector((store) => store.user.errors)
+  const accessToken = useSelector(store => store.user.accessToken)
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  // every time accessToken is changing from our redux store we should redirect user to '/' path
+  useEffect(() => {
+    if (accessToken) {
+      history.push('/')
+    }
+  }, [accessToken, history])
+
+  let loginErrorMessage = ''
+  let emailErrorMessage = ''
+  if (errors?.error?.errors?.email?.message) {
+    emailErrorMessage = errors?.error?.errors?.email?.message //email validation
+    } else if (errors?.message) {
+      loginErrorMessage = errors?.message //login failed
+  } 
```

Comment:
Awesome that you figured out a solution for the email error message! 🥳 Could you maybe add a comment to help explain this code for others reading your code?